### PR TITLE
Update Permissions for JDA5

### PR DESCRIPTION
**Pull request by DManstrator**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [x] Internal code
- [x] Library interface (affecting end-user code) 
- [x] Documentation
- [ ] Other: \_____ 

## Description

Added, removed and changed several Permissions.

### Removed:
- Removed `Permission.MESSAGE_READ` and used the existing `Permission.VIEW_CHANNEL` instead.

### Added:
- Added `Permission.MESSAGE_SEND_IN_THREADS`.

### Renamed:
- `Permission.USE_SLASH_COMMANDS` was renamed to `Permission.USE_APPLICATION_COMMANDS`.
- `Permission.MANAGE_EMOTES` was renamed to `Permission.MANAGE_EMOTES_AND_STICKERS`.
- `Permission.MESSAGE_WRITE` was renamed to `Permission.MESSAGE_SEND`.
- `Permission.USE_PUBLIC_THREADS` was renamed to `Permission.CREATE_PUBLIC_THREADS`.
- `Permission.USE_PRIVATE_THREADS` was renamed to `Permission.CREATE_PRIVATE_THREADS`.

Notes for users (can be copy pasted for Discord):
```
---
**`Permission.MESSAGE_READ` was removed** since it is the old version of **`Permission.VIEW_CHANNEL`, use that instead**.
**`Permission.MESSAGE_SEND_IN_THREADS` was added**. With that you can control which Member / Role can send Messages in Threads.
**`Permission.USE_SLASH_COMMANDS`** was renamed to **`Permission.USE_APPLICATION_COMMANDS`**.
**`Permission.MANAGE_EMOTES`** was renamed to **`Permission.MANAGE_EMOTES_AND_STICKERS`**.
**`Permission.MESSAGE_WRITE`** was renamed to **`Permission.MESSAGE_SEND`**.
**`Permission.USE_PUBLIC_THREADS`** was renamed to **`Permission.CREATE_PUBLIC_THREADS`**.
**`Permission.USE_PRIVATE_THREADS`** was renamed to **`Permission.CREATE_PRIVATE_THREADS`**.
```

### CODE REVIEW

**Comment by DManstrator** [comment]
@DV8FromTheWorld When looking at the permissions, I saw that `Read Text Channels & See Voice Channels` is called `View Channels` in the UI too (on Server, Channel and Category) and it belongs to the General Server Permissions. Should I change that too? I would also reorder the permissions to match the current UI.

File: src/main/java/net/dv8tion/jda/internal/entities/TextChannelImpl.java [hunk 1]
```diff
@@ -407,7 +407,7 @@ public MessageAction sendFile(@Nonnull byte[] data, @Nonnull String fileName, @N
     @Override
     public RestAction<Void> pinMessageById(@Nonnull String messageId)
     {
-        checkPermission(Permission.MESSAGE_READ, "You cannot pin a message in a channel you can't access. (MESSAGE_READ)");
+        checkPermission(Permission.VIEW_CHANNEL, "You cannot pin a message in a channel you can't access. (MESSAGE_READ)");
```
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/internal/entities/TextChannelImpl.java:410]: These error messages still reference `MESSAGE_READ`.
Please replace all references to `MESSAGE_READ` throughout the code.

**Review by DV8FromTheWorld — CHANGES_REQUESTED** [review]

**Comment by DManstrator** [comment]
Waiting for #1827 before merging now.

**Comment by DManstrator** [comment]
@DV8FromTheWorld I am done with that PR for now, let me know if you want something to be changed.

**Review by DV8FromTheWorld — APPROVED** [review]
This seems fine to me.

**Pull request merged**